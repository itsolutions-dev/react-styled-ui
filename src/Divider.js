import styled, { css } from './styledComponent';
import { getTextColor } from './utils/';

const Divider = styled({
  component: 'hr',
  style: css`
    border: 0;
    border-bottom-width: ${props => props.size !== undefined ? props.size : '1px'};
    border-bottom-style: solid;
    border-bottom-color: ${props => props.color !== undefined ? props.color : getTextColor(props)};
  `,
});

export default Divider;
