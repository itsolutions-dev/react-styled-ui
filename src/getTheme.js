import defaultTheme from './theme';
import deepMerge from './utils/deepMerge';

const getTheme = deepMerge.bind(null, defaultTheme);

export default getTheme;
