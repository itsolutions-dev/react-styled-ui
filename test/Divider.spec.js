import React from 'react';
import ReactTestRenderer from 'react-test-renderer';
import 'styled-components-test-utils/lib/jest';
import getTheme from '../src/getTheme';
import Divider from '../src/Divider';

const theme = getTheme({
  palette: {
    defaultText: '#C0C0C0',
  },
});

describe('Divider', () => {
  test('should render a Divider', () => {
    const component = ReactTestRenderer.create(<Divider theme={theme} />);
    expect(component).toBeDefined();
    expect(component.toJSON()).toMatchSnapshot();
  });

  test('should have an hr tag', () => {
    const component = ReactTestRenderer.create(<Divider theme={theme} />).toJSON();
    expect(component.type).toEqual('hr');
  });

  test('should have size 1px by default', () => {
    const component = ReactTestRenderer.create(
      <Divider
        theme={theme}
      />,
    );
    expect(component).toHaveStyleRule('border-bottom-width', '1px');
  });

  test('should handle size prop', () => {
    const component = ReactTestRenderer.create(
      <Divider
        theme={theme}
        size="2px"
      />,
    );
    expect(component).toHaveStyleRule('border-bottom-width', '2px');
  });

  test('should have a default color', () => {
    const component = ReactTestRenderer.create(
      <Divider
        theme={theme}
      />,
    );
    expect(component).toHaveStyleRule('border-bottom-color', '#C0C0C0');
  });

  test('should handle color prop', () => {
    const component = ReactTestRenderer.create(
      <Divider
        theme={theme}
        color="blue"
      />,
    );
    expect(component).toHaveStyleRule('border-bottom-color', 'blue');
  });
});
