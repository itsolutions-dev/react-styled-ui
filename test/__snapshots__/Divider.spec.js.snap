// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Divider should render a Divider 1`] = `
.c0 {
  border: 0;
  border-bottom-width: 1px;
  border-bottom-style: solid;
  border-bottom-color: #C0C0C0;
}

<hr
  className="c0"
/>
`;
